### ISSUE TITLE
Extraneous double-quotes rendered in 'foreach' Ergo Expression

### ISSUE THREAD

**Issue by martinhalford** [accordproject/ergo#768]
**Describe the bug**
Ergo `foreach` renders extra double-quotes within a `grammer.tem.md`

**To Reproduce**
Create an `asset` such as follows:
```
asset ExclusiveSaleAuthorityContract extends AccordContract {
  ...
  o Commission[] commissions
}
```
Model the Commission as follows:
```
concept Commission {
  o Double percentAmount
  o Double dollarAmount
  o Double price
  o String description
}
```
In the data.json file, have an array of commissions like follows...
```
  "commissions" : [
    {
      "$class" : "systems.benext.reiv.Commission",
      "percentAmount" : 7,
      "dollarAmount" : 35000,
      "price" : 1650000,
      "description" : "7% of the property sale between $0 and $500,000"
    },
    {
      "$class" : "systems.benext.reiv.Commission",
      "percentAmount" : 7.0,
      "dollarAmount" : 20000,
      "price" : 1500000,
      "description" : "8% of the property sale between $500,000 and $1,000,000"
    }
  ]
```
Create the following Ergo expression in a `grammar.tem.md`...
```
{{%
    foreach c in commissions
    return "$ " ++ c.dollarAmount as "0,0" ++ "\n"
%}}
{{%
    foreach c in commissions
    return c.percentAmount as "0,0.0" ++ " percent" ++ "\n"
%}}
{{%
    foreach c in commissions
    return "$ " ++ c.price as "0,0" ++ "\n"
%}}
{{%
    foreach c in commissions
    return c.description ++ "\n"
%}}
```
You will get something similar to the following:
```
$ 20,000
""$ 35,000
7.0 percent
""7.0 percent
$ 1,500,000
""$ 1,650,000
7% of the property sale between $0 and $500,000
""8% of the property sale between $500,000 and $1,000,000
```

**Expected behavior**

Expect no extra "" quote marks.
```
$ 20,000
$ 35,000
7.0 percent
7.0 percent
$ 1,500,000
$ 1,650,000
7% of the property sale between $0 and $500,000
8% of the property sale between $500,000 and $1,000,000
```

**Additional context**
Cicero 0.21.3

**Comment by jeromesimeon** [comment]
Some more data on this.
1. Ergo returns the proper array of lists
<img width="686" alt="Screenshot 2020-08-14 at 9 42 06 AM" src="https://user-images.githubusercontent.com/670099/90255880-e7ebf680-de12-11ea-8a67-697fb65bdf79.png">
2. It's likely an artefact of how arrays are turned into text when Ergo is used in formulas (I'll look at a fix, but not sure what people would really expect, maybe it should be a type error?)
3. You should be able to fix your code to avoid that issue by constructing the string result yourself. e.g.,:
```
{{%
    join("\n", foreach c in commissions return "$ " ++ c.dollarAmount as "0,0")
%}}
```

**Comment by jeromesimeon** [comment]
4. I think this code would look nicer like as a `join` template block:
```
{{#join commissions separator=""}}$ {{dollarAmount as "0,0"}}
{{/join}}

**Comment by jeromesimeon** [comment]
Example in template studio. With this grammar:
<img width="1014" alt="Screenshot 2020-08-14 at 9 58 36 AM" src="https://user-images.githubusercontent.com/670099/90257170-c8ee6400-de14-11ea-834d-e84e019da5f3.png">

Get this drafted text:
<img width="1007" alt="Screenshot 2020-08-14 at 9 59 16 AM" src="https://user-images.githubusercontent.com/670099/90257196-d4da2600-de14-11ea-86bb-3d8c8ce30e1f.png">

**Comment by martinhalford** [comment]
Thanks Jerome. I'll try the join operator, as suggested.

**Comment by jeromesimeon** [comment]
Should be fixed in #776 . Here is a test with volume discount. For the following grammar:
```
bash-3.2$ more text/grammar.tem.md
#### Discount.

The Discount is determined according to the following table:
{{#ulist rates}}
{{volumeAbove}}$ million <= Volume < {{volumeUpTo}}$ million : {{rate}}%
{{/ulist}}

{{%
foreach r in rates return
"[rate is " ++ toString(r.rate) ++ "] "
%}}
```

Before the fix:
```
bash-3.2$ cicero draft 
#### Discount.

The Discount is determined according to the following table:
-  0.0$ million <= Volume < 1.0$ million : 3.1%
-  1.0$ million <= Volume < 10.0$ million : 3.1%
-  10.0$ million <= Volume < 50.0$ million : 2.9%
-  50.0$ million <= Volume < 500.0$ million : 2.5%
-  500.0$ million <= Volume < 1000.0$ million : 1.2%
-  1000.0$ million <= Volume < 1000000.0$ million : 0.1%

{{%"[rate is 0.1] ""[rate is 1.2] ""[rate is 2.5] ""[rate is 2.9] ""[rate is 3.1] ""[rate is 3.1] "%}}
```
After the fix:
```
bash-3.2$ ~/git/cicero/packages/cicero-cli/index.js draft 
#### Discount.

The Discount is determined according to the following table:
-  0.0$ million <= Volume < 1.0$ million : 3.1%
-  1.0$ million <= Volume < 10.0$ million : 3.1%
-  10.0$ million <= Volume < 50.0$ million : 2.9%
-  50.0$ million <= Volume < 500.0$ million : 2.5%
-  500.0$ million <= Volume < 1000.0$ million : 1.2%
-  1000.0$ million <= Volume < 1000000.0$ million : 0.1%

{{%[rate is 0.1] [rate is 1.2] [rate is 2.5] [rate is 2.9] [rate is 3.1] [rate is 3.1] %}}
```

This fix will be released in Cicero `0.21.4`

**Comment by jeromesimeon** [comment]
Cicero `0.21.4` has been published with a fix for this. Closing this as resolved. Please reopen (or open a new issue) if further problems.
